index.bs

```diff
@@ -11705,6 +11710,81 @@ The [=remote end steps=] given |session| and |command parameters| are:
 
 </div>
 
+### Events ### {#module-input-events}
+
+#### The input.fileDialogOpened Event #### {#event-input-fileDialogOpened}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl remote-cddl">
+         input.FileDialogOpened = (
+            method: "input.fileDialogOpened",
+            params: input.FileDialogInfo
+         )
+
+         input.FileDialogInfo = {
+            context: browsingContext.BrowsingContext,
+            element: script.SharedReference,
+            multiple: bool,
+         }
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm>
+The [=remote end event trigger=] is the
+<dfn export>WebDriver BiDi file dialog shown</dfn> steps, given |element|.
+
+1. Assert |element| implements {{HTMLInputElement}}.
+
+1. Let |navigable| be the |element|'s [=node document=]'s [=/navigable=].
+
+1. Let |navigable id| be |navigable|'s [=navigable id=].
+
+1. Let |multiple| be <code>true</code> if |element|'s <{input/multiple}> attribute is
+   set, or <code>false</code> otherwise.
+
+1. Let |related navigables| be a [=/set=] containing |navigable|.
+
+1. For each |session| in the [=set of sessions for which an event is enabled=] given
+   "<code>input.fileDialogOpened</code>" and |related navigables|:
+
+  1. Let |shared id| be [=get shared id for a node=] with |element| and |session|.
+
+  1. Let |params| be a [=/map=] matching the <code>input.FileDialogInfo</code>
+     production with the <code>context</code> field set to |navigable id|, the
+     <code>element</code> field set to |shared id| and <code>multiple</code> field
+     set to |multiple|.
+
+  1. Let |body| be a [=/map=] matching the <code>input.fileDialogOpened</code>
+     production, with the <code>params</code> field set to |params|.
+
+  1. [=Emit an event=] with |session| and |body|.
+
+1. Let |dismissed| be false.
+
+1. 1. For each |session| in [=active BiDI sessions=]:
+
+  1. Let |user prompt handler| be |session|'s [=user prompt handler=].
+
+  1. If |user prompt handler| is not null:
+
+   1. Assert |user prompt handler| is a [=map=].
+
+   1. If |user prompt handler| [=map/contains=] "<code>fileDialog</code>":
+
+      1. if |user prompt handler|["<code>fileDialog</code>"] is not equal to
+         "<code>ignore</code>", set |dismissed| to true.
+
+   1. Otherwise if |user prompt handler| [=map/contains=] "<code>default</code>" and
+      |user prompt handler|["<code>fileDialog</code>"] is not equal to
+      "<code>ignore</code>", set |dismissed| to true.
```

Comment:
Respecting default behavior makes sense, but is a breaking change, as default of "accept" and "dismiss" starts suppressing file dialog.